pkg/metrics/cluster.go

```diff
@@ -0,0 +1,136 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+	utilmetrics "github.com/karmada-io/karmada/pkg/util/metrics"
+)
+
+const (
+	clusterReadyMetricsName              = "cluster_ready_info"
+	clusterTotalNodeNumberMetricsName    = "cluster_node_number"
+	clusterReadyNodeNumberMetricsName    = "cluster_ready_node_number"
+	clusterMemoryAllocatableMetricsName  = "cluster_memory_allocatable_bytes"
+	clusterCPUAllocatableMetricsName     = "cluster_cpu_allocatable_number"
+	clusterPodAllocatableMetricsName     = "cluster_pod_allocatable_number"
+	clusterMemoryAllocatedMetricsName    = "cluster_memory_allocated_bytes"
+	clusterCPUAllocatedMetricsName       = "cluster_cpu_allocated_number"
+	clusterPodAllocatedMetricsName       = "cluster_pod_allocated_number"
+	clusterSyncStatusDurationMetricsName = "cluster_sync_status_duration"
+)
+
+var (
+	// clusterReadyGauge reports if the cluster is ready.
+	clusterReadyGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: clusterReadyMetricsName,
+		Help: "if cluster is ready.",
+	}, []string{"cluster_name", "type"})
```

Comment:
```suggestion
	}, []string{"cluster_name", "sync_mode"})
```
or 
```suggestion
	}, []string{"cluster_name", "mode"})
```
Same as below.